aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -1045,6 +1070,17 @@ impl VMAdapter for AptosVM {
                 let (vm_status, output) =
                     self.execute_user_transaction(data_cache, txn, log_context);
 
+                if let Err(DiscardedVMStatus::UNKNOWN_INVARIANT_VIOLATION_ERROR) =
+                    vm_status.clone().keep_or_discard()
```

Comment:
Good. We should do this way more extensively (e.g. have counters etc. for all unexpected VM violations, there are more codes than this one.) At least this is a start.